src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -157,13 +156,13 @@ def variant_compatible_siblings(filenames, variant=None) -> Union[List[os.PathLi
     transformers_index_format = r"\d{5}-of-\d{5}"
 
     if variant is not None:
-        # `diffusion_pytorch_model.fp16.bin` as well as `model.fp16-00001-of-00002.safetensors`
+        # `diffusion_pytorch_model.fp16.bin` as well as `model-00001-of-00002.fp16.safetensors`
         variant_file_re = re.compile(
-            rf"({'|'.join(weight_prefixes)})\.({variant}|{variant}-{transformers_index_format})\.({'|'.join(weight_suffixs)})$"
+            rf"({'|'.join(weight_prefixes)})(-{transformers_index_format})?\.{variant}\.({'|'.join(weight_suffixs)})$"
         )
```

Comment:
When the checkpoints are serialized in the format mentioned in the comments, the regex also needs to be modified accordingly.